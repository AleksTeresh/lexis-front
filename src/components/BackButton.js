import React from 'react'
import Button from 'material-ui/Button'
import ArrowBackIcon from 'material-ui-icons/ArrowBack'
import styled from 'styled-components'

type ActionButtonProps = {
  color?: string,
  ariaLabel?: string,
  onClick?: () => void,
  style?: any
}
export default function BackButton ({
  color = 'primary',
  ariaLabel = 'back',
  onClick = () => {},
  style = {}
}: ActionButtonProps) {
  return (
    <Button
      raised
      color={color}
      aria-label={ariaLabel}
      onClick={onClick}
      style={{
        margin: '15px 15px 15px 15px',
        ...style
      }}
    >
      <ArrowBackIcon />
    </Button>
  )
}
