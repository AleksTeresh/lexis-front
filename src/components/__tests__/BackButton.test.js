/* eslint-env jest */

import React from 'react'
import BackButton from '../BackButton'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'

configure({ adapter: new Adapter() })

describe('BackButton Component', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(<BackButton />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
