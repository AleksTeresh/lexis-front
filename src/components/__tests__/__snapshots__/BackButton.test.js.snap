// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton Component renders a snapshot 1`] = `
<withStyles(Button)
  aria-label="back"
  color="primary"
  onClick={[Function]}
  raised={true}
  style={
    Object {
      "margin": "15px 15px 15px 15px",
    }
  }
>
  <pure(ArrowBack) />
</withStyles(Button)>
`;
